### ISSUE TITLE
testing: drainAll to allow returning promise from eventBus.flush

### ISSUE THREAD

**Issue by x1B** [LaxarJS/laxar#431]
Sometimes, a call to `eventBus.flush` triggers asynchronous but immediate operations that would cause multiple event bus ticks in an application. For example, axVisibility and its mock work asynchronously. In tests, we currently have to use constructs such as these:

```js
axMock.eventBus.flush();
Promise.resolve().then( () => {
   axMock.eventBus.flush();
   done();
} );
```

These heavily depend on the promise/event-bus interaction details. In these cases, one would wish for a flush that does "everything that happens immediately".

So, the semantics of the proposed new `eventBus.drainAsync` is:
 - immediately return a promise that will only be resolved when there are no more event-bus events after a `Promise.resolve()` tick

This will allow to rewrite the above as follows:

```js
axMock.eventBus.drainAsync().then( done );
```

This will leave the existing `eventBus.flush` as-is.

**Comment by x1B** [comment]
_edited:_ proposed a backward-compatible solution

**Comment by x1B** [comment]
Implemented on `master` (`v2.0.0`).

### Upgrade Information

A method `drainAsync` was added to the mock event bus (usually accessed as through [laxar-mocks](https://github.com/LaxarJS/laxar-mocks) as `axMocks.eventBus`).

The new method is similar to `flush()`, but instead of simulating an event bus tick synchronously, it returns a promise that will be fulfilled after all event bus callbacks have been run asynchronously, even new callbacks that were added asynchronously.

Previously, some tests (or beforeEach/afterEach blocks) had to use awkward cascades of `eventBus.flush()`:

```js
axMock.eventBus.flush();
Promise.resolve().then( () => {
   axMock.eventBus.flush();
   done();
} );
```

These can now be rewritten as follows:

```js
axMock.eventBus.drainAsync().then( done );
```

This will leave the existing `eventBus.flush` as-is.
